bundle/ScheduledVisibility/Handler/Location.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Bundle\IbexaScheduledVisibilityBundle\ScheduledVisibility\Handler;
+
+use Ibexa\Contracts\Core\Repository\ContentService;
+use Ibexa\Contracts\Core\Repository\Exceptions\BadStateException;
+use Ibexa\Contracts\Core\Repository\LocationService;
+use Ibexa\Contracts\Core\Repository\Repository;
+use Ibexa\Contracts\Core\Repository\Values\Content\Content;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Enums\HandlerType;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\ScheduledVisibility\ScheduledVisibilityInterface;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+
+use function method_exists;
+use function sprintf;
+
+final class Location implements ScheduledVisibilityInterface
+{
+    public function __construct(
+        private readonly Repository $repository,
+        private readonly ContentService $contentService,
+        private readonly LocationService $locationService,
+        private readonly LoggerInterface $logger = new NullLogger(),
+    ) {}
+
+    public function hide(Content $content): void
+    {
+        try {
+            if ($this->isHidden($content)) {
+                return;
+            }
+        } catch (BadStateException $e) {
```

Comment:
`BadStateException` here can come from a `Content` that is not yet published. Since we are sourcing `Content` from search, which only works on published `Content`, and from after publish event, I would not try to handle that case; we can let it fail.

Reply:
fixed in ace2cb9758143cc21d56e0fcbc99474a371a7ae8